db/migrate/20170726174757_recover_bios_tokens_statuses.rb

```diff
@@ -0,0 +1,27 @@
+
+# will only work in Ruby 1.9+:
+class String
+   def multibyte?
+     chars.count < bytes.count
+   end
+end
+
+class RecoverBiosTokensStatuses < ActiveRecord::Migration
+  def up
+      DrupalUsers.find(:all).each do |u|
+        unless u.name.multibyte?
+          user = u.user
+          if user
+            user.status = u.status unless u.nil?
+            user.bio = DrupalProfileValue.find_by_uid(user.id, conditions: { fid: 7 }) || ''
```

Comment:
user.id ? :-( now everyone has numeric bios! :-p That explains it. This took so long that I'm tempted to use `execute` to run a SQL query directly.